xrootd/internal/xrdenc/xrdenc.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package xrdenc // import "go-hep.org/x/hep/xrootd/internal/xrdenc"
+
+import "encoding/binary"
+
+// Encoder encodes values to a buffer according to the XRootD protocol.
+type Encoder struct {
+	buf []byte
+}
+
+func (enc *Encoder) Bytes() []byte { return enc.buf }
+
+func (enc *Encoder) WriteU8(v uint8) {
+	enc.buf = append(enc.buf, v)
+}
+
+func (enc *Encoder) WriteU16(v uint16) {
+	var buf [2]byte
+	binary.BigEndian.PutUint16(buf[:], v)
+	enc.buf = append(enc.buf, buf[:]...)
+}
+
+func (enc *Encoder) WriteI32(v int32) {
+	var buf [4]byte
+	binary.BigEndian.PutUint32(buf[:], uint32(v))
+	enc.buf = append(enc.buf, buf[:]...)
+}
+
+func (enc *Encoder) WriteI64(v int64) {
+	var buf [8]byte
+	binary.BigEndian.PutUint64(buf[:], uint64(v))
+	enc.buf = append(enc.buf, buf[:]...)
+}
+
+func (enc *Encoder) WriteBool(v bool) {
+	if v {
+		enc.buf = append(enc.buf, 1)
+	} else {
+		enc.buf = append(enc.buf, 0)
+	}
+}
+
+func (enc *Encoder) WriteLen(n int) {
+	enc.WriteI32(int32(n))
+}
+
+func (enc *Encoder) WriteBytes(vs []byte) {
+	enc.buf = append(enc.buf, vs...)
+}
+
+func (enc *Encoder) WriteStr(str string) {
+	enc.WriteLen(len(str))
+	enc.WriteBytes([]byte(str))
+}
+
+func (enc *Encoder) WriteReserved(n int) {
```

Comment:
looking back at this, it seems I wasn't completely super inspired when coming up with `WriteReserved` as a name.

Perhaps `WritePadding` ?
(not completely happy with that either... what about you?)

Reply:
What do you think about `Skip` cause what we really do is skipping some of the bytes? However, I'm not sure how it fits in with other `WriteXXX` methods...

Reply:
If we go with the RBuffer, then I'd be tempted to rename it as 'Next', to follow 'bytes.Buffer' lead.